### PR TITLE
UIU-3224 Implement duplicates users view

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://folio-org.atlassian.net/browse/UIU-3224

Display a list of duplicates for pre-stage patron user.

---
The first part of UIU-3224 - #2769 

## Approach

 - Create a separate route for viewing duplicates of pre-stage users;
 - An autosized MCL will occupy the entire height of the parent Pane element, which has a set pixel value. Since, in addition to the MCL, there is also a banner inside the Pane, it is necessary to calculate the height of the container in which the MCL will be stored without an unnecessary scrollbar. For this, ResizeObserver should be used.
-  To preserve information about filters between pages, we will use location state. Since the user information page can be opened in a new tab from the duplicates page, for better UX, you should check the "referrer" property when closing the page. For security, you should also make sure the links comply with the same origin policy.

## Screenshot

https://github.com/user-attachments/assets/69075248-64b8-4beb-8481-bb77680f9a4a

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by Terala-Priyanka** [comment]
Hey @usavkov-epam 
This feature is for LOC which do not have ECS env. So, I think we need not bother about tenantId in the fetch calls.

**Comment by usavkov-epam** [comment]
> Hey @usavkov-epam This feature is for LOC which do not have ECS env. So, I think we need not bother about tenantId in the fetch calls.

I used a common pattern for common hooks that can get `tenantId`. Just because the current function doesn't need to care about tenants doesn't mean that another part of the code won't :)

**Comment by usavkov-epam** [comment]
@folio-org/fe-tl-reviewers please

**Review by NikitaSedyx — APPROVED** [review]

**Review by alisher-epam — APPROVED** [review]

File: src/hooks/usePatronGroups/usePatronGroups.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+const queryClient = new QueryClient();
+const wrapper = ({ children }) => (
+  <QueryClientProvider client={queryClient}>
+    {children}
+  </QueryClientProvider>
+);
+
+const usergroups = [{ id: 'ug-1' }];
+
+describe('useUserAffiliations', () => {
```
Comment (Terala-Priyanka) [inline src/hooks/usePatronGroups/usePatronGroups.test.js:23]: Rename the test suite

Comment (usavkov-epam) [reply src/hooks/usePatronGroups/usePatronGroups.test.js]: done

**Comment by Terala-Priyanka** [comment]
Please update the permission name at https://github.com/folio-org/ui-users/blob/3f3d25b9e7c7cbb1bcf91fb8f630b882b046d162/src/components/LinkToPatronPreRegistrations/LinkToPatronPreRegistrations.js#L15, without which the feature will not be testable, after merge

**Comment by usavkov-epam** [comment]
> Please update the permission name at
> 
> https://github.com/folio-org/ui-users/blob/3f3d25b9e7c7cbb1bcf91fb8f630b882b046d162/src/components/LinkToPatronPreRegistrations/LinkToPatronPreRegistrations.js#L15
> 
> , without which the feature will not be testable, after merge

Done

**Review by Terala-Priyanka — APPROVED** [review]
Thank you!

**Pull request merged**